### PR TITLE
feat: Basic landing site (#7)

### LINKED ISSUE #7: [Feat]: Create basic landing site
### My suggestion

Using tailwind design landing page with "GwardiaHub" text and short project description.
It should be nicely centered.

### Possible alternatives

Design even better landing page!

_Mod's note: Requires #5 to be closed._

### PR DESCRIPTION
**Pull request by Sanielia**
Created basic landing site with flexbox layout

### CODE REVIEW

**Review by Norbiros — CHANGES_REQUESTED** [review]
1. Ymm, I don't like that there are a lot of spammed margins with random values, margin auto etc.
2. You could change old "Gwardia Czapli" message, because it doesn't fit imo (for example remove red colour etc.)
3. Also using 3 different level of margin doesn't look quite good
image

**Comment by goteusz-maszyk** [comment]
> 2. You could change old "Gwardia Czapli" message, because it doesn't fit imo (for example remove red colour etc.)

He deleted this i think

File: core/templates/core/index.html [hunk 1]
```diff
@@ -3,5 +3,16 @@
 {% block title %}Strona Główna{% endblock %}
 
 {% block content %}
-    <p class="text-3xl font-bold underline text-red-600">Gwardia Czapli</p>
+    <div class="flex flex-col flex-nowrap items-center">
+        <h1 class="text-4xl font-bold underline text-red-600 mt-10 mb-16">Gwardia Czapli</h1>
```
Comment (goteusz-maszyk) [inline core/templates/core/index.html]: Remove this

File: core/templates/core/index.html [hunk 2]
```diff
@@ -3,5 +3,16 @@
...
+        <div class="flex flex-col flex-nowrap items-start mr-auto ml-16">
+            <h2 class="text-3xl -ml-4">Gwardia Hub website</h2>
```
Comment (goteusz-maszyk) [inline core/templates/core/index.html]: This can be centered

**Review by goteusz-maszyk — CHANGES_REQUESTED** [review]
Also I lost my admin

**Comment by goteusz-maszyk** [comment]
Oh and some background gradient would be nice

**Comment by Sanielia** [comment]
> 1. Ymm, I don't like that there are a lot of spammed margins with random values, margin auto etc.
> 2. You could change old "Gwardia Czapli" message, because it doesn't fit imo (for example remove red colour etc.)
> 3. Also using 3 different level of margin doesn't look quite good
>    image

1. ml-auto/mr-auto is to center/justify element to one of sides, flex doesn't have justify-self, but I will change it, beacuse there is no reason for adjusting flex to just h1 element, instead of adjusting h1 element to other flex. Also I don't really understand now, why I did it in first place, because on flex-col cross-axis is horizontal and align-self is used by flex
2. I will change it to Gwardia Hub and change h2 to Project goal
3. Imo using 3 diffrents level of margin, where 1 is for header, one for paragraph and one for list is better. It can look strange, when you have only one of each, but when we will add more paragraphs and headers I am certain, that it will look better, that displaying all in one line

File: core/templates/core/index.html (hunk 1 above)
Comment (Sanielia) [reply core/templates/core/index.html]: Will change h1 to Gwardia Hub and h2 to Project goal

File: core/templates/core/index.html (hunk 2 above)
Comment (Sanielia) [reply core/templates/core/index.html]: I think article header should be left-aligned, it just looks better, but I will change this h2 to Project goal

**Comment by Sanielia** [comment]
> Oh and some background gradient would be nice

I don't really like gradients. Maybe just plain background?

**Review by Norbiros — CHANGES_REQUESTED** [review]
Use Polish https://github.com/Gwardia-Czapli/GwardiaHub/discussions/17

File: static/GwardiaHub/tailwind_custom_styles.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Used for custom styles in templates
+
+tailwind.config = {
+      theme: {
+        extend: {
+          colors: {
+            main_bg: '#0d1017',
+            border_bg: '#484f58',
```
Comment (goteusz-maszyk) [inline static/GwardiaHub/tailwind_custom_styles.js]: this border is quite ugly...

**Review by goteusz-maszyk — CHANGES_REQUESTED** [review]

File: static/GwardiaHub/tailwind_custom_styles.js (hunk 3 above)
Comment (Sanielia) [reply static/GwardiaHub/tailwind_custom_styles.js]: You mean its color?

Comment (goteusz-maszyk) [reply static/GwardiaHub/tailwind_custom_styles.js]: like what's the purpose

Comment (Sanielia) [reply static/GwardiaHub/tailwind_custom_styles.js]: Welp, IMO background with border looks better than without

File: core/templates/core/index.html [hunk 4]
```diff
@@ -3,5 +3,16 @@
...
 {% block content %}
-    <p class="text-3xl font-bold underline text-red-600">Gwardia Czapli</p>
+    <div class="flex items-center justify-center h-screen text-white">
+        <div class="text-center">
+            <h1 class="text-4xl text-white font-bold my-5">Gwardia Hub</h1>
+            <span>Strona internetowa, która ma służyć pomocą szkolną oraz efektywnym zarządzaniem aktywnościami Gwardii.</span><br/>
+            <span class="font-bold">W trakcie programowania!</span>
+            <br><br>
+            <div class="items-center">
+                <a href="https://example.com" class="hover:text-blue-700 text-gray-400 transition-colors duration-300"><i class="fa-brands fa-github fa-3x mr-5"></i></a>
```
Comment (Norbiros) [inline core/templates/core/index.html]: Change to correct link. Maybe modify hover color? Move `<i>` to other line that `<a>`

File: core/templates/core/index.html [hunk 5]
```diff
@@ -3,5 +3,16 @@
...
+                <a href="https://example.com" class="hover:text-blue-500 text-gray-400 transition-colors duration-300"><i class="fa-brands fa-discord fa-3x"></i></a>
```
Comment (Norbiros) [inline core/templates/core/index.html]: Same as above.

File: static/GwardiaHub/tailwind_custom_styles.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+            header_text: '#afbac5',
+            basic_text: '#afbac5',
```
Comment (Norbiros) [inline static/GwardiaHub/tailwind_custom_styles.js:9]: Can you remove unused colors?

**Review by Norbiros — CHANGES_REQUESTED** [review]

File: core/templates/core/index.html (hunk 4 above)
Comment (Kubaryt) [reply core/templates/core/index.html:13]: I literally copied you code btw xd

File: static/GwardiaHub/tailwind_custom_styles.js (hunk 6 above)
Comment (Sanielia) [reply static/GwardiaHub/tailwind_custom_styles.js:9]: RIght

File: core/templates/core/base.html [hunk 7]
```diff
@@ -4,12 +4,41 @@
         <meta charset="UTF-8">
         <meta name="viewport" content="width=device-width, initial-scale=1.0">
         <script src="https://cdn.tailwindcss.com"></script>
+        {% load static %}
+        <script src="{% static "tailwind_custom_styles.js" %}"></script>
+        <script src="https://unpkg.com/starback@2.1.1/dist/starback.global.js"></script>
         <title>GwardiaHub | {% block title %}{% endblock %}</title>
```
Comment (Norbiros) [inline core/templates/core/base.html]: Move title above other scripts.

File: core/templates/core/base.html [hunk 8]
```diff
@@ -4,12 +4,41 @@
...
+
+        <link href="{% static 'fontawesomefree/css/fontawesome.css' %}" rel="stylesheet" type="text/css">
+        <link href="{% static 'fontawesomefree/css/brands.css' %}" rel="stylesheet" type="text/css">
+        <link href="{% static 'fontawesomefree/css/solid.css' %}" rel="stylesheet" type="text/css">
     </head>
-    <body>
+    <body class="bg-main_bg">
         <main>
+            <canvas id="canvas" class="fixed -z-1"></canvas>
```
Comment (Norbiros) [inline core/templates/core/base.html]: Don't use id like this

File: core/templates/core/base.html [hunk 9]
```diff
@@ -4,12 +4,41 @@
...
+        <script>
+            const canvas = document.getElementById('canvas');
+            const starback = new Starback(canvas, {
+                type: 'dot',
+                quantity: 100,
+                height: 600,
+                width: 800,
+                backgroundColor: [],
+                randomOpacity: true,
+                // Add more options as needed
```
Comment (Norbiros) [inline core/templates/core/base.html]: You could remove this comment

File: core/templates/core/index.html [hunk 10]
```diff
@@ -3,5 +3,20 @@
...
 {% block content %}
-    <p class="text-3xl font-bold underline text-red-600">Gwardia Czapli</p>
+    <div class="flex items-center justify-center h-screen text-basic_text">
+        <div class="text-center">
+            <h1 class="text-4xl text-header_text font-bold my-5">Gwardia Hub</h1>
+            <span>Strona internetowa, która ma służyć pomocą szkolną oraz efektywnym zarządzaniem aktywnościami Gwardii.</span><br/>
+            <span class="font-bold">W trakcie programowania!</span>
+            <br><br>
+            <div class="items-center">
+                <a href="https://example.com" class="hover:text-blue-700 text-gray-400 transition-colors duration-300">
```
Comment (Norbiros) [inline core/templates/core/index.html]: Fix links

**Review by Norbiros — CHANGES_REQUESTED** [review]

File: core/templates/core/base.html (hunk 8 above)
Comment (Sanielia) [reply core/templates/core/base.html]: What do you mean?

File: core/templates/core/base.html (hunk 9 above)
Comment (Sanielia) [reply core/templates/core/base.html]: Right

File: core/templates/core/index.html (hunk 10 above)
Comment (Sanielia) [reply core/templates/core/index.html]: I forgor

File: core/templates/core/base.html (hunk 8 above)
Comment (Norbiros) [reply core/templates/core/base.html]: Id `canvas` isn't descriptive. You could use `document.querySelector("canvas")` (search by HTML tag). Or jusr change id

Comment (Sanielia) [reply core/templates/core/base.html]: `starback-bg` as id would be good enough?

**Comment by Norbiros** [comment]
But website looks awesome!

File: core/templates/core/base.html [hunk 11]
```diff
@@ -3,13 +3,40 @@
...
+            const starback = new Starback(canvas, {
+                type: 'dot',
+                quantity: 250,
+                height: window.innerHeight,
+                width: window.innerWidth,
+                backgroundColor: [],
+                randomOpacity: true,
+            });
+            function resizeCanvas() {
+            canvas.width = window.innerWidth;
```
Comment (Norbiros) [inline core/templates/core/base.html]: Same as above.

File: core/templates/core/base.html [hunk 12]
```diff
@@ -3,13 +3,40 @@
...
+        <link href="{% static 'fontawesomefree/css/fontawesome.css' %}" rel="stylesheet" type="text/css">
+        <link href="{% static 'fontawesomefree/css/brands.css' %}" rel="stylesheet" type="text/css">
+        <link href="{% static 'fontawesomefree/css/solid.css' %}" rel="stylesheet" type="text/css">
     </head>
-    <body>
+    <body class="bg-main_bg">
         <main>
+            <canvas id="starback-bg" class="fixed -z-1"></canvas>
+            <div class="relative z-1">
             {% block content %}
```
Comment (Norbiros) [inline core/templates/core/base.html]: This needs to be indented (add one tab):

**Review by Norbiros — CHANGES_REQUESTED** [review]

**Review by Norbiros — APPROVED** [review]

**Review by goteusz-maszyk — APPROVED** [review]

**Pull request merged**